packages/ai/src/ui/abstract-chat.ts

```diff
@@ -158,53 +181,69 @@ export abstract class AbstractChat<
     this.messageMetadataSchema = messageMetadataSchema;
     this.dataPartSchemas = dataPartSchemas;
     this.state = state;
+    this.onError = onError;
+    this.onToolCall = onToolCall;
+    this.onFinish = onFinish;
   }
 
-  getStatus(): ChatStatus {
-    return this.state.getStatus();
+  /**
+   * Hook status:
+   *
+   * - `submitted`: The message has been sent to the API and we're awaiting the start of the response stream.
+   * - `streaming`: The response is actively streaming in from the API, receiving chunks of data.
+   * - `ready`: The full response has been received and processed; a new user message can be submitted.
+   * - `error`: An error occurred during the API request, preventing successful completion.
+   */
+  get status(): ChatStatus {
+    return this.state.status;
   }
 
-  setStatus({ status, error }: { status: ChatStatus; error?: Error }) {
-    if (this.getStatus() === status) return;
+  protected setStatus({
+    status,
+    error,
+  }: {
+    status: ChatStatus;
+    error?: Error;
+  }) {
+    if (this.status === status) return;
 
-    this.state.setStatus(status);
-    this.state.setError(error);
+    this.state.status = status;
+    this.state.error = error;
 
     this.emit({ type: 'status-changed' });
   }
 
-  getError() {
-    return this.state.getError();
+  get error() {
+    return this.state.error;
   }
 
-  getMessages() {
-    return this.state.getMessages();
+  get messages() {
+    return this.state.messages;
   }
 
-  getLastMessage() {
-    return this.state.getMessages()[this.state.getMessages().length - 1];
+  get lastMessage():
```

Comment:
added `undefined` to the type as well as making it a getter, so it works with type narrowing now and fixes a couple of potential bugs